clusterupgrade/cluster_upgrade_worker.go

```diff
@@ -823,9 +867,26 @@ func checkAgentImageAgainstStatusFile(workDir string) (bool, error) {
 	}
 }
 
+func checkAndUpdateImagePullSecret(kubeClient *KubeClient, nmpName string, imagePullValMap map[string]string) error {
+	// update image pull secret
+	glog.Infof(cuwlog(fmt.Sprintf("Updating image pull secret for nmp %v", nmpName)))
+
+	irHost := os.Getenv(AGENT_CLUSTER_IMAGE_REGISTRY_HOST)
```

Comment:
the value should be already fetched in `imagePullValMap`

Reply:
Not really. It is in the `imagePullValMap` only if the image registry host is put in the config file